src/Widgets/HeaderBar.vala

```diff
@@ -75,19 +65,27 @@ public class Camera.Widgets.HeaderBar : Gtk.HeaderBar {
             warning ("Styling take button failed: %s", e.message);
         }
 
-        Gtk.StyleContext take_button_style_context = take_button.get_style_context ();
+        var take_button_style_context = take_button.get_style_context ();
         take_button_style_context.add_provider (take_button_style_provider, Gtk.STYLE_PROVIDER_PRIORITY_APPLICATION);
         take_button_style_context.add_class ("take-button");
         take_button_style_context.add_class (Gtk.STYLE_CLASS_DESTRUCTIVE_ACTION);
-        take_button_style_context.add_class (Gtk.STYLE_CLASS_RAISED);
 
         mode_button = new Granite.Widgets.ModeButton ();
         mode_button.append_icon (PHOTO_ICON_SYMBOLIC, Gtk.IconSize.BUTTON);
         mode_button.append_icon (VIDEO_ICON_SYMBOLIC, Gtk.IconSize.BUTTON);
         mode_button.sensitive = false;
 
-        this.set_custom_title (take_button);
-        this.pack_end (mode_button);
+        show_close_button = true;
+        set_custom_title (take_button);
+        pack_end (mode_button);
+
+        update_take_button_icon ();
+        connect_signals ();
+    }
+
+    public void set_is_recording (bool is_recording) {
```

Comment:
I didn't mess with is_recording because it seems to be broken for me and I didn't want the git blame to point to me before someone fixes it :p